core/src/main/scala/org/apache/spark/status/api/v1/ApiRootResource.scala

```diff
@@ -193,6 +206,13 @@ private[spark] trait UIRoot {
   def getSparkUI(appKey: String): Option[SparkUI]
   def getApplicationInfoList: Iterator[ApplicationInfo]
 
+  def writeEventLogs(appId: String, attemptId: Option[String], zipStream: ZipOutputStream): Unit = {
```

Comment:
This needs a java doc since it's technically not the same as the one you defined in `ApplicationHistoryProvider`. We should expand on why attemptId is an option (what happens if it's `None`).